core_lib/src/structure/filemanager.cpp

```diff
@@ -25,25 +25,26 @@ GNU General Public License for more details.
 #include "object.h"
 #include "layercamera.h"
 
-
-QString openErrorTitle = QObject::tr("Could not open file");
-QString openErrorDesc = QObject::tr("There was an error processing your file. This usually means that your project has "
-                         "been at least partially corrupted. You can try again with a newer version of Pencil2D, "
-                         "or you can try to use a backup file if you have one. If you contact us through one of "
-                         "our official channels we may be able to help you. For reporting issues, "
-                         "the best places to reach us are:");
-QString contactLinks = "<ul>"
-                       "<li><a href=\"https://discuss.pencil2d.org/c/bugs\">Pencil2D Forum</a></li>"
-                       "<li><a href=\"https://github.com/pencil2d/pencil/issues/new\">Github</a></li>"
-                       "<li><a href=\"https://discord.gg/8FxdV2g\">Discord<\a></li>"
-                       "</ul>";
-
+namespace
+{
+    QString openErrorTitle = QObject::tr("Could not open file");
+    QString openErrorDesc = QObject::tr("There was an error processing your file. This usually means that your project has "
+                             "been at least partially corrupted. You can try again with a newer version of Pencil2D, "
+                             "or you can try to use a backup file if you have one. If you contact us through one of "
+                             "our official channels we may be able to help you. For reporting issues, "
+                             "the best places to reach us are:");
+    QString contactLinks = "<ul>"
+                           "<li><a href=\"https://discuss.pencil2d.org/c/bugs\">Pencil2D Forum</a></li>"
+                           "<li><a href=\"https://github.com/pencil2d/pencil/issues/new\">Github</a></li>"
+                           "<li><a href=\"https://discord.gg/8FxdV2g\">Discord<\a></li>"
+                           "</ul>";
+}
```

Comment:
I don't think so. Here's some discussion about the same error: https://www.reddit.com/r/C_Programming/comments/2nkpcq/understanding_clang_wmissingvariabledeclarations/. Basically it sounds like it's to avoid conflicts in the global variable namespace. For example, if two files define contactLinks with different values in the file scope like this, then what should the value be?

I just learned about anonymous namespaces which make the variables only accessible from filemanager.cpp. This seemed like a convenient solution to me. These variables could have also been static members of FileManager, or declared extern in the header file which, from what I understand, does a check to make sure that it is only defined once externally.